torch/linalg/__init__.py

```diff
@@ -88,6 +88,49 @@
 Alias of :func:`torch.det`.
 """)
 
+matrix_rank = _add_docstr(_linalg.linalg_matrix_rank, r"""
+matrix_rank(input, tol=None, hermitian=False) -> Tensor
+
+Computes the numerical rank of a matrix :attr:`input`,
+or of each matrix in a batched :attr:`input`. The method to compute the
+matrix rank is done using SVD by default. If :attr:`hermitian` is ``True``,
+then :attr:`input` is assumed to be Hermitian (symmetric if real-valued),
+and the computation of the rank is done by obtaining the eigenvalues.
+
+:attr:`tol` is the threshold below which the singular values (or the eigenvalues
+when :attr:`hermitian` is ``True``) are considered to be 0. If :attr:`tol` is not
+specified, :attr:`tol` is set to ``S.max() * max(S.size()) * eps`` where `S` is the
+singular values (or the eigenvalues when :attr:`hermitian` is ``True``), and ``eps``
+is the epsilon value for the datatype of :attr:`input`.
+
+Supports input of ``float``, ``double``, ``cfloat`` and ``cdouble`` datatypes.
+
```

Comment:
This function will cause a cross-device memory sync because of the functions it calls internally, right?

If so, we should add a note that this will occur like we've been doing in other linalg functions.

cc @heitorschueroff